olive/passes/onnx/peephole_optimizer.py

```diff
@@ -28,6 +31,19 @@ def __init__(self, source_model_path):
         self.source_model_path = str(source_model_path)
         self.model = onnx.load(self.source_model_path)
 
+    def remove_initializer_from_input(self) -> onnx.ModelProto:
```

Comment:
is there a use case for this transformation. the same name being input and initializer makes the input optional for most ort EPs. this can be intentional. 
For example, extract adapter pass adds initializers to use a default value for optional inputs. 
Maybe this can be moved into the graph surgery as a configurable option?

Reply:
Discussed offline. Move this to graph surgery pass.